core/src/test/scala/unit/kafka/server/ApiVersionsRequestTest.scala

```diff
@@ -0,0 +1,41 @@
+/**
+  * Licensed to the Apache Software Foundation (ASF) under one or more
+  * contributor license agreements.  See the NOTICE file distributed with
+  * this work for additional information regarding copyright ownership.
+  * The ASF licenses this file to You under the Apache License, Version 2.0
+  * (the "License"); you may not use this file except in compliance with
+  * the License.  You may obtain a copy of the License at
+  *
+  *    http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+
+package kafka.server
+
+import org.apache.kafka.common.protocol.ApiKeys
+import org.apache.kafka.common.requests.{ApiVersionsRequest, ApiVersionsResponse}
+import org.junit.Assert._
+import org.junit.Test
+
+class ApiVersionsRequestTest extends BaseRequestTest {
+
+  override def numBrokers(): Int = 1
+
+  @Test
+  def testApiVersionsRequest() {
+    val apiVersionsResponse = sendApiVersionsRequest(new ApiVersionsRequest, 0)
+
+    assertEquals("API keys in ApiVersionsResponse must match API keys supported by broker.", ApiKeys.values.length, apiVersionsResponse.apiVersions.size)
+    assertTrue("API versions in ApiVersionsResponse must be supported by broker.", KafkaApis.apiVersionsResponse.apiVersions().containsAll(apiVersionsResponse.apiVersions()))
```

Comment:
I know. The issue is that if you define `equals`, you also should also define `hashCode`. And maybe `toString`. And if someone uses `ApiVersionResponse.ApiVersion` and notices that those methods are defined, they will probably expect the same for classes in other response classes.

In my opinion, we should see these classes as the classes generated by protocol buffers, thrift, etc. They should be as consistent as possible (a bit sad that we have to do it manually).